src/audio/AudioEngine.h

```diff
@@ -1,66 +1,175 @@
 #pragma once
 
-#include <string>
+#include <cstddef>
+#include <cstdint>
+#include <functional>
+#include <memory>
 #include <vector>
 
-#include <glm/glm.hpp>
-
-typedef struct ALCdevice_struct ALCdevice;
+#include <math/mathlib.h>
 
 namespace Audio
 {
-    class AudioWorld;
+    enum class State {
+        Stopped,
+        Paused,
+        Playing
+    };
 
-    /** The AudioEngine represents the target OS sound system.
-     *
-     * The engine class matches the OpenAL device level.
+    enum class Format {
+        Mono,
+        Stereo
+    };
+
+    /**
+     * @brief Sounds represent audio sources.
+     * 
+     * All of the information about a sound should be re-set before it is played
+     * after it had been stopped.
+     * 
+     */
+    class Sound {
+    public:
```

Comment:
Missing virtual destructor.

Reply:
I have to admit, I am not sure why a virtual destructor is needed here 😨

Reply:
Consider
```
Sound *s = new MyCoolSound();
delete s;
```
Without virtual destructor only base class destructor will be called so every resource (like memory) claimed by derivative class will not be released.